pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -37,6 +37,18 @@ func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clien
 	cfg := config.FromContextOrDefaults(ctx)
 	pr := pipelineRun.Spec.PipelineRef
 	namespace := pipelineRun.Namespace
+	// if the spec is already in the status, do not try to fetch it again, just use it as source of truth
+	if pipelineRun.Status.PipelineSpec != nil {
+		return func(_ context.Context, name string) (v1beta1.PipelineObject, error) {
+			return &v1beta1.Pipeline{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+				Spec: *pipelineRun.Status.PipelineSpec,
+			}, nil
+		}, nil
+	}
```

Comment:
can we please add a unit test for this? 🥺

Reply:
So this function is implemented to fetch the definition from the cluster or bundle reference but we are checking if the pipeline is already running i.e. status is not nil, return the stored specifications irrespective of bundle or no bundle.

Reply:
done 😉 
Yes, the pipeline doesn't really have to run though, just the pipelineSpec in status being populated